### ISSUE TITLE
Documentation about how it works the collaboration feature between computers

### ISSUE THREAD

**Issue by driucorado** [zed-industries/zed#8260]
### Check for existing issues

- [X] Completed

### Describe the feature

I want to have more information about how the collaboration works, in terms of security to see if makes sense to me and my company to use it. Also if is possible to host the servers in our own infraestructure. or how the collaboration works in general. 

I am someone working as system architecture and security advicer so i would like to know more about it. (like more in detail documentation about it) 

Loving the tool!

**Comment by ConradIrwin** [comment]
@driucorado you can read some information here: https://zed.dev/docs/channels; and a summary of what information is sent here: https://zed.dev/faq#data-and-privacy.

We build zed using zed, and there are other companies already using it internally too. If you have specific concerns or worries, I'd love to talk them through; but we've built this in a way where we hope companies feel able to run it on codebases that they do not wish to be public.

The code for the collab server is in `crates/collab` if you'd like to audit it.

We do "technically" allow hosting your own, but this is not well supported at the moment. At a minimum you will need:
* To host the server itself + a Postgres database on your servers
* To reimplement the authentication flow using a website that is not https://zed.dev
* To abide by the AGPL license for any changes you make to the editor.
* set the `"server_url"` configuration in the editor.

**Comment by Qkessler** [comment]
Hi there @ConradIrwin !

> To reimplement the authentication flow using a website that is not https://zed.dev/

Particularly interested on this one. Mind diving deeper into how the flow works and how should interested parties go about reimplementing the authentication flow?

```mermaid
sequenceDiagram
    ZedInstance->>ZedServerURL: /native_app_signin?native_app_port={}&native_app_public_key={}
    ZedServerURL->>ZedServerURL: What is this doing here?
    ZedServerURL->>ZedInstance: ...?access_token=a&user_id=b
    ZedInstance->>ZedServerURL: Location: /native_app_signin_succeeded
```

Specially interested in how the access token is computed. Assuming Github is contacted but is the access token one generated by Github? From the code, the access_token is returned base64 encoded with the public key, decoded with the private key on the Zed instance.

Last one: What is the `access_token` used for? Is there documentation around that I can check? I see the `access_tokens` are saved on the Collab DB:
```
CREATE TABLE IF NOT EXISTS "access_tokens" (
    "id" SERIAL PRIMARY KEY,
    "user_id" INTEGER REFERENCES users (id),
    "hash" VARCHAR(128)
);
```
[source](https://github.com/zed-industries/zed/blob/ee207ab77ecd61fad41bb6191e978898dbfae2db/crates/collab/migrations/20210607190313_create_access_tokens.sql#L1)

Thanks!
Best,
Quique.

**Comment by Qkessler** [comment]
Have a follow-up question: How're the access token pulled with the Sign In button and the one mentioned on `collab` connected? Isn't the created by the Collaboration Server ([ref](https://github.com/zed-industries/zed/blob/ee207ab77ecd61fad41bb6191e978898dbfae2db/crates/collab/src/auth.rs#L128-L137)) and then verified on each request to it through the `Authorization` header?

**Comment by ConradIrwin** [comment]
Currently zed.dev runs an Oauth flow with Github.com, and then calls [`get_authenticated_user`](https://github.com/zed-industries/zed/blob/2af9fa7785a0db2aefd24f816bb20d05e9ea9dda/crates/collab/src/api.rs#L156) and then calls into `create_access_token` as you pointed out.

Happy to spend some time with you talking this through, or getting it working if you'd like: https://calendar.app.google/TmQ3LR9gU6DqHXD9A

I'd love to make sure we have a good story for people who want either:
* To work seamlessly with GitHub Enterprise (probably an easy step 1)
* To work with a different auth provider (maybe a bit less easy of a step 2).

**Comment by Qkessler** [comment]
Another idea: We can standardize the way the collab server communicates with the Zed instance. Say, with `zsh://` deep links. We can create a "protocol" to 1. authenticate 2. return the information required by the Zed instance. For example, the log in, the profile picture and so on.

**Comment by ConradIrwin** [comment]
Using http links allows us to more easily support setups where people have multiple release-channels of Zed installed (zed:// may not go to the running Zed instance if you have Zed preview and Zed Stable installed);

I think it'd be nice to see if we can scope the changes so we only need changes to collab to make this work (but we may have to do some groundwork on the client to make that possible). Things like AI billing probably become a non-issue if you're running your own collab server, so we could figure out how to disable those features as part of it.

**Comment by Qkessler** [comment]
> Using http links allows us to more easily support setups where people have multiple release-channels of Zed installed (zed:// may not go to the running Zed instance if you have Zed preview and Zed Stable installed);

Good point, happy to stick to HTTP at the moment, don't think it's a blocker.

**Comment by Qkessler** [comment]
I'm starting to look into the code to find patterns we can abstract. I'll soon set up something with you @ConradIrwin . Created a quick PR to update some local set up instructions.

**Comment by ConradIrwin** [comment]
Great, looking forward to chatting! I’m out next week, so may have to be early January

**Comment by JWesorick** [comment]
This is real old but wanted to add. My team would love to use channels but as long as they are hosted on zed servers we won't be able to.

**Comment by luckylinux** [comment]
I tried to put some Notes together as to be honest there was a bit of reverse engineering of the Server, how to setup LiveKit & Amazon-S3 Compatible Storage, Reverse Proxy, Authentication, etc.

https://github.com/luckylinux/zed-collab-resources

Unfortunately it does **NOT** work yet since OIDC Login doesn't work (zed-server replies with 404 at `https://zed.MYDOMAIN.TLD/api/native_app_signin`) and I haven't got around that yet.

Contributions are welcome, though it would be much easier if there was official Documentation provided by the Project Developers.

**Comment by monjaris** [comment]
how my collab peer will compile my project after we both editing it?
